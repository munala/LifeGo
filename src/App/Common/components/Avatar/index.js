import React from 'react';
import PropTypes from 'prop-types';
import Avatar from '@material-ui/core/Avatar';

import './styles.css';

const UserAvatar = ({
  src,
  name,
  style,
  onClick,
}) => (
  <div
    className="avatar-container"
    onClick={onClick}
  >
    <Avatar
      alt={name}
      src={src}
      style={style}
    />
  </div>
);

UserAvatar.propTypes = {
  src: PropTypes.string.isRequired,
  name: PropTypes.string,
  style: PropTypes.shape({}),
  onClick: PropTypes.func,
};

UserAvatar.defaultProps = {
  name: 'user avatar',
  style: undefined,
  onClick: () => {},
};

export default UserAvatar;
