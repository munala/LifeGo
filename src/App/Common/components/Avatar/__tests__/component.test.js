import React from 'react';
import { shallow } from 'enzyme';
import Avatar from '../';

describe('Avatar tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Avatar src="source" />);
    expect(wrapper).toMatchSnapshot();
  });
});
