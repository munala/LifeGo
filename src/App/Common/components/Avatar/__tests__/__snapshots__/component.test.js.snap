// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar tests renders without crashing 1`] = `
<div
  className="avatar-container"
  onClick={[Function]}
>
  <WithStyles(Avatar)
    alt="user avatar"
    src="source"
  />
</div>
`;
